pages/doc.js

```diff
@@ -94,48 +77,90 @@ export default class Documentation extends Component {
     })
   }
 
-  getLinkHref = (section, file) => {
-    const sectionSlug =
-      sidebar[section].slug || kebabCase(sidebar[section].name)
-    const fileSlug = file ? kebabCase(file.slice(0, -3)) : undefined
-    return `/doc/${compact([sectionSlug, fileSlug]).join('/')}`
+  getLinkHref = (section, subsection = null, file = null) => {
+    let sect = sidebar[section]
+    let removeExtFunc = filename =>
+      SidebarMenuHelper.removeExtensionFromFileName(filename)
+    const sectionSlug = removeExtFunc(sect.indexFile) || kebabCase(sect.name)
+    const subsectionSlug =
+      (subsection && removeExtFunc(sect.files[subsection].indexFile)) ||
+      sect.files[subsection]
+    const fileSlug = removeExtFunc(file) || (file && file.files[0])
+    return `${PATH_SEPARATOR}doc${PATH_SEPARATOR}${compact([
+      sectionSlug,
+      subsectionSlug,
+      fileSlug
+    ]).join(PATH_SEPARATOR)}`
   }
 
-  setCurrentPath = (section, file) => {
-    window.history.pushState(null, null, this.getLinkHref(section, file))
+  setCurrentPath = (section, subsection, file) => {
+    window.history.pushState(
+      null,
+      null,
+      this.getLinkHref(section, subsection, file)
+    )
   }
 
   onSectionSelect = (section, e) => {
     e && e.preventDefault()
-    const { indexFile, files } = sidebar[section]
-    const file = indexFile || flatten(files)[0]
-    e && this.setCurrentPath(section, indexFile ? undefined : file)
-    this.loadFile({ file, section, parseHeadings: false })
+    const file = sidebar[section].indexFile || sidebar[section].files[0]
+    e && this.setCurrentPath(section)
+    this.loadFile({ section, file, parseHeadings: false })
   }
 
-  onFileSelect = (file, section, e) => {
+  onFileSelect = (section, subsection, file, e) => {
     e && e.preventDefault()
-    this.setCurrentPath(section, file)
-    this.loadFile({ file, section, parseHeadings: true })
+    this.setCurrentPath(
+      section,
+      subsection,
+      file.indexFile ? file.indexFile : file
```

Comment:
same ... is file or not? what is the reason to pass different structures around?

Reply:
fix it